pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -129,23 +130,40 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
 		return cmdutil.UsageError(cmd, "Must pass a name of a build config or specify build name with '--from-build' flag")
 	}
 
-	name := buildName
-	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
-		isBuild = false
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
 	}
 
-	if webhooks.Provided() {
-		return RunListBuildWebHooks(f, out, cmd.Out(), name, isBuild, webhooks.String())
+	var (
+		name     = buildName
+		resource = "builds"
+	)
+
+	if len(name) == 0 && len(args) > 0 && len(args[0]) > 0 {
+		mapper, _ := f.Object()
+		resource, name, err = osutil.ResolveResource("buildconfigs", args[0], mapper)
+		if err != nil {
+			return err
+		}
+		if len(name) == 0 {
```

Comment:
move this check outside the if block to catch cases where len(name)==0, but len(args[0]) == 0, for example